### PR TITLE
[feature] Add Virtual DNS Support

### PR DESCRIPTION
**Pull request by justinbatcf**

### CODE REVIEW

File: virtualdns.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
+package cloudflare
+
+import (
+    "encoding/json"
+    "github.com/pkg/errors"
+)
+
+/*
+Create a Virtual DNS cluster
```
Comment (elithrar) [inline virtualdns.go]: Can you change this to mirror the (golint approved) comment style? Refer to https://github.com/cloudflare/cloudflare-go/blob/master/dns.go#L10-L13

File: cloudflare.go [hunk 2]
```diff
@@ -405,3 +405,28 @@ type IPsResponse struct {
...
+    Name                 string   `json:"name"`
+    OriginIPs            []string `json:"origin_ips"`
+    VirtualDNSIPs        []string `json:"virtual_dns_ips"`
+    MinimumCacheTTL      uint     `json:"minimum_cache_ttl"`
+    MaximumCacheTTL      uint     `json:"maximum_cache_ttl"`
+    DeprecateAnyRequests bool     `json:"deprecate_any_requests"`
+    ModifiedOn           string   `json:"modified_on"`
+}
+
+type VirtualDNSResponse struct {
```
Comment (elithrar) [inline cloudflare.go]: Update this to look like the below:

```go
type VirtualDNSResponse struct {
    Response
    Result VirtualDNS `json:"result"`
}
```

As per: https://github.com/cloudflare/cloudflare-go/blob/master/cloudflare.go#L152-L155

File: cloudflare.go [hunk 3]
```diff
@@ -405,3 +405,28 @@ type IPsResponse struct {
...
+    Success  bool         `json:"success"`
+    Errors   []string     `json:"errors"`
+    Messages []string     `json:"messages"`
+    Result   VirtualDNS `json:"result"`
+}
+
+type VirtualDNSListResponse struct {
```
Comment (elithrar) [inline cloudflare.go]: As above.

**Comment by elithrar** [comment]
Thanks @justinbatcf - just a couple of minor comments. You'll also need to rebase against master as well (sorry!).

**Comment by Tenzer** [comment]
A round of `gofmt` will also help please the test robots.

File: cloudflare.go (hunk 2 above)
Comment (Tenzer) [inline cloudflare.go]: Also, as per #49 it would be great if you moved this and the following struct into `virtualdns.go`.

**Comment by jamesog** [comment]
You've managed to commit the `flarectl` binary - can you remove this please?

**Comment by Tenzer** [comment]
Since I can't do an in-line comment on a binary file: Please don't commit the compiled version of `flarectl`, it's currently located in `cmd/flarectl/flarectl` in your branch.

File: virtualdns.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+API reference:
+  https://api.cloudflare.com/#virtual-dns-users--create-a-virtual-dns-cluster
+  POST /user/virtual_dns
+*/
+func (api *API) CreateVirtualDNS(v VirtualDNS) (VirtualDNS, error) {
+	res, err := api.makeRequest("POST", "/user/virtual_dns", v)
+	if err != nil {
+		return VirtualDNS{}, errors.Wrap(err, errMakeRequestError)
+	}
+
```
Comment (Tenzer) [inline virtualdns.go]: Please remove this extra blank line and the one on line 26 for consistency.

**Comment by elithrar** [comment]
Follow https://github.com/cloudflare/cloudflare-go/pull/75

**Pull request closed without merging**